Reneiski_Homework1/ReneiskiSelenium11/Program.cs

```diff
@@ -0,0 +1,239 @@
+    {
+        private IWebDriver _driver;        
+        private WebDriverWait _driverWait;
+        private Actions _driverActions;
+
+        [OneTimeSetUp]
+        public void OneTimeSetup()
+        {
+            _driver = new ChromeDriver();
+            _driver.Manage().Window.Maximize();
+            _driverWait = new WebDriverWait(_driver, TimeSpan.FromSeconds(1000));
+            _driverActions = new Actions(_driver);
+        }
+
+        [Test]
+        public void FirstCheckBoxTest()
+        {
+            _driver.Navigate().GoToUrl("https://demoqa.com/checkbox");
+
+            var firstCheckBox = _driver.FindElement(By.XPath("//*[@class='rct-icon rct-icon-uncheck']"));
+            firstCheckBox.Click();
+
+            var checkedCheckbox = _driver.FindElement(By.XPath("//*[@class='rct-icon rct-icon-check']"));
+            var isCheckboxChecked = checkedCheckbox.GetAttribute("class").Contains("check");
+            
+            Assert.IsTrue(isCheckboxChecked);
+
+            var currentResult = _driver.FindElement(By.Id("result"));
+            var currentResultText = currentResult.Text;
+            
+            Assert.AreEqual(currentResultText, ("You have selected :\r\nhome\r\ndesktop\r\nnotes\r\ncommands\r\ndocuments\r\nworkspace\r\nreact\r\nangular\r\nveu\r\noffice\r\npublic\r\nprivate\r\nclassified\r\ngeneral\r\ndownloads\r\nwordFile\r\nexcelFile"));
+        }
+
+        [Test]
+        public void SecondRadioButtonTest()
+        {
+            _driver.Navigate().GoToUrl("https://demoqa.com/radio-button");
+
+            var radioButtonYes = _driver.FindElement(By.XPath("//*[@for='yesRadio']"));
+            radioButtonYes.Click();
+
+            var currentResult = _driver.FindElement(By.XPath("//*[@class='mt-3']"));
+            var currentResultText = currentResult.Text;
+
+            Assert.AreEqual(currentResultText, "You have selected Yes");            
+
+            var radioButtonImpressive = _driver.FindElement(By.XPath("//*[@for='impressiveRadio']"));
+            radioButtonImpressive.Click();
+
+            var currentResult1 = _driver.FindElement(By.XPath("//*[@class='mt-3']"));
+            var currentResultText1 = currentResult1.Text;
+
+            Assert.AreEqual(currentResultText1, "You have selected Impressive");            
+        }
+
+        [Test]
+        public void ThirdWebTablesTest()
+        {
+            var firstName = "Alex";
+            var lastName = "Ren";
+            var userEmail = "Ren@tut.by";
+            var userAge = "41";
+            var userSalary = "20000";
+            var userDepartment = "QA";
+            var editedUserSalary = "22000";            
+
+            _driver.Navigate().GoToUrl("https://demoqa.com/webtables");
+
+            var currentCierra = _driver.FindElement(By.XPath("(//*[@class=\"rt-tr-group\"]//div[@role='row'])[1]//div[@role='gridcell'][count(//*[contains(text(),'First Name')]//ancestor::div[1]//preceding-sibling::div)]"));
```

Comment:
this locator is pretty the same for all columns, but the only thing that is changing is column name
you may create a method on a test class level that will have columnName as parameter, will find element depends on column name and return this element

Reply:
Fixed